catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeMapImpl.java

```diff
@@ -100,6 +102,26 @@ public String getValue() {
     return Objects.toString(json.get(VALUE));
   }
 
+  @Override
+  public Double getDistance() {
+    if (!isLeaf()) {
+      throw new IllegalStateException("No target value exists for a logical operator");
+    }
+
+    Object distance = json.get(DISTANCE);
+    if (distance == null) {
+      return null;
+    } else if (distance instanceof Double) {
+      return (Double) distance;
+    } else if (distance instanceof Integer) {
+      return ((Integer) distance).doubleValue();
+    } else if (distance instanceof Long) {
+      return ((Long) distance).doubleValue();
+    } else {
+      throw new IllegalStateException("The distance value could not be converted into a double");
+    }
+  }
```

Comment:
❓ Do we need to support all of these? Or is just `Double` sufficient? I thought incoming filter JSON should, for the `distance` attribute, always be a `Double`.

Reply:
Unfortunately, the normalization logic is necessary